File: lab-2.1/c++/Ironman.cpp
#include "Ironman.h"

Ironman::Ironman() : Human(), suitModel("Mark I"), powerLevel(100) {}

Ironman::Ironman(std::string n, int a, std::string sm, int pl) 
    : Human(n, a), suitModel(sm), powerLevel(pl) {}

Ironman::Ironman(const Ironman& other) 
    : Human(other), suitModel(other.suitModel), powerLevel(other.powerLevel) {}

std::string Ironman::getSuitModel() const {
    return suitModel;
}

int Ironman::getPowerLevel() const {
    return powerLevel;
}

File: lab-2.1/c#/Ironman.cs
namespace CharacterLib
{
    public class Ironman : Human
    {
        private string suitModel;
        private int powerLevel;

        // Конструктори
        public Ironman() : base()
        {
            suitModel = "Mark I";
            powerLevel = 100;
        }

        public Ironman(string name, int age, string suitModel, int powerLevel) 
            : base(name, age)
        {
            this.suitModel = suitModel;
            this.powerLevel = powerLevel;
        }

        public Ironman(Ironman other) : base(other)
        {
            suitModel = other.suitModel;
            powerLevel = other.powerLevel;
        }

        // Гетери
        public string GetSuitModel()
        {
            return suitModel;
        }

        public int GetPowerLevel()
        {
            return powerLevel;
        }
    }
}